neqo-transport/src/connection/mod.rs

```diff
@@ -2122,10 +2133,10 @@ impl Connection {
     fn set_initial_limits(&mut self) {
         let tps = self.tps.borrow();
         let remote = tps.remote();
-        self.indexes.remote_max_stream_bidi =
-            StreamIndex::new(remote.get_integer(tparams::INITIAL_MAX_STREAMS_BIDI));
-        self.indexes.remote_max_stream_uni =
-            StreamIndex::new(remote.get_integer(tparams::INITIAL_MAX_STREAMS_UNI));
+        let _ = self.local_streams_fc[StreamType::BiDi]
+            .update(remote.get_integer(tparams::INITIAL_MAX_STREAMS_BIDI));
+        let _ = self.local_streams_fc[StreamType::UniDi]
+            .update(remote.get_integer(tparams::INITIAL_MAX_STREAMS_UNI));
```

Comment:
I couldn't work this out for a while, but then I realized that the remote streams object was returning the wrapper type, but the local streams one returns the underlying type.  This might be confusing.  At least until you get the Deref thing sorted out properly, I guess.